File: CS/MvpvmNavigation/ViewModels/ModuleB/ModuleBViewModel.cs
using System;
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using MvpvmNavigation.ViewModels;

namespace MvpvmNavigation.ViewModels {
    public class ModuleBViewModel {
        [ServiceProperty]
        public virtual IDocumentManagerService DocumentManagerService {
            get { throw new NotImplementedException(); }
        }
        [Command]
        public void ShowEditForm() {
            var document = DocumentManagerService.CreateDocument("EditForm", null, this);
            if (document != null) {
                EditFormViewModel viewModel = document.Content as EditFormViewModel;
                if (viewModel != null)
                    viewModel.Document = document;
                document.Show();
            }
        }
    }
}

File: VB/MvpvmNavigation/ViewModels/ModuleB/ModuleBViewModel.vb
Imports System
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports MvpvmNavigation.ViewModels

Namespace MvpvmNavigation.ViewModels
    Public Class ModuleBViewModel
        <ServiceProperty>
        Public Overridable ReadOnly Property DocumentManagerService() As IDocumentManagerService
            Get
                Throw New NotImplementedException()
            End Get
        End Property
        <Command>
        Public Sub ShowEditForm()
            Dim document = DocumentManagerService.CreateDocument("EditForm", Nothing, Me)
            If document IsNot Nothing Then
                Dim viewModel As EditFormViewModel = TryCast(document.Content, EditFormViewModel)
                If viewModel IsNot Nothing Then
                    viewModel.Document = document
                End If
                document.Show()
            End If
        End Sub
    End Class
End Namespace